### ISSUE TITLE
[A11Y] Screen Reader is interrupted on new messages / submitted message is read twice

### ISSUE THREAD

**Issue by cimbis** [microsoft/BotFramework-WebChat#3390]
As can be seen in [screen recording](https://youtu.be/6nIB7j-TxbU), after submitting a message, it's being read out twice, and when the bot replies the screen reader is interrupted with the new message.

Test flow - https://microsoft.github.io/BotFramework-WebChat/05.custom-components/c.user-highlighting/ 

I am experiencing similar issue with VoiceOver + Safari in the solution I am developing. Am looking for any kind of hints to solve this - I think that each of the messages should be read only once and whenever a new message is posted, screen reader should announce it only after it has read out the queue.

**Comment by compulim** [comment]
This is interesting. Let me investigate and get back to you shortly.

**Comment by compulim** [comment]
I am also able to repro it on Safari on macOS using a standard W3C sample at https://www.w3.org/WAI/WCAG21/working-examples/aria-role-log/chatlog.html. This sample show how chat should behave in an accessible environment. I have included a video here.

[Screen Recording 2020-08-10 at 3.09.52 PM.zip](https://github.com/microsoft/BotFramework-WebChat/files/5053681/Screen.Recording.2020-08-10.at.3.09.52.PM.zip)

If you listen carefully, you will notice:

- "Hi!" is partially skipped
- "Please to meet you." is skipped because it is interrupted by "We'll talk later, okay?"
- The "okay" in "We'll talk later, okay?" is interrupted by "Bye!"

Unfortunately this is a browser-specific issue. Web Chat can't workaround this problem because screen reader works transparently in the browser. We cannot monitor/control how the screen reader works. We can't "wait until the text finished narrating" then we add content to the screen reader.

Could you file a bug to Safari to see if they can help?

**Comment by cimbis** [comment]
Hi @compulim!

I've tried the chatlog example from w3 as well as looked at the screen recording you provided. 
I did not experience interruptions upon new messages, as they appeared on screen after VoiceOver had read them (default "rate" setting). What seems to me is that your VoiceOver is set to read text slower ("rate" is reduced) and that's why it was possible to replicate the issue. I was able to achieve the same result as well, once the "rate" setting was reduced on my VoiceOver instance.

However, I tried to replicate the issue on [FireFox](https://youtu.be/rkO_8TUwVoA) as well as [Chrome](https://youtu.be/p5hGqDJ1NB8). As it can be seen, FireFox and VoiceOver combination isn't doing a good job, however Chrome + VoiceOver acts in a similar manner as Safari + VoiceOver. So there is something more to Safari being the source of issue, at least I'd like to think so.

I've read quite some articles regarding screen reading support and stuff related to that and pretty much everywhere it says that VoiceOver and Safari are good with respecting `role="log" aria-live="polite"`. The w3 chatlog example doesn't have the `aria-live="polite"` tag attached to the live region, so maybe that's the spot were the issue lies...?

Also, upon investigating the architecture of WebChat I've noticed that there are 2 live regions implemented (again using the same example WebChat as above):
![log_x2](https://user-images.githubusercontent.com/5424191/90119147-de568600-dd61-11ea-92f0-48ee7b1b0041.png) 
I'm wondering whether the two regions could conflict with each other in some way and be the cause for such behaviour of screen reader?

**Comment by compulim** [comment]
Hi @cimbis. Thanks for investigating the issue.

I tune it slower so W3C sample will be slow enough to get one sentence overlapping another. With standard speed, the sentence was too short to get overlapped.

We have 2 `<div>` with `aria-live="polite"`. The first one is [toast notification](https://microsoft.github.io/BotFramework-WebChat/05.custom-components/i.notification/). The second one is the transcript (i.e. messages).

During our repro, no toasts were popped (in other words, nothing is added to that `<div>`). So I think it's safe to say the first one is not interfering with VoiceOver on incoming messages.

**Comment by cimbis** [comment]
I'm out of ideas here.. What's your opinion - is there anything I can do to fix the matter?

**Comment by compulim** [comment]
As we demonstrated the issue with W3C WCAG 2.1 sample, it seems Safari has a bug on `live="region"` and causing the speech to be interrupted. FYI, the sample is a golden standard on how various accessibility feature should work. And seems they failed the bar.

As this problem live outside of Web Chat, there are limited options we could do here:

1. File a bug to Safari team
1. Find a workaround

For workaround, we are happy to take any standard HTML + JS that could exhibit that it won't trigger the issue. Then, we can investigate the legitimacy of the workaround and downfalls, and we port the implementation to Web Chat.

Does it sound good?

**Comment by compulim** [comment]
If you do not agree W3C WCAG sample is good for benchmarking, can you come up with a HTML + JS sample that would work on Safari + VoiceOver and Chrome + ChromeVox? We can adopt the approach.

The sample need to be simple and exhibit the problem:

- Simple
   1. Using `<ul>` and `<li>`, or `<div>` only
   1. Only use `setTimeout` to mimic incoming messages, do not fetch over the network. This is to reduce chances of false negatives
- Exhibit the problem
   1. Exhibit the interruption by having a very long narration, followed by another narration

**Comment by lzc850612** [comment]
Closing due to inactivity. @cimbis  feel free to reopen if needed.
